MekHQ/src/mekhq/campaign/finances/CurrencyManager.java

```diff
@@ -116,17 +120,26 @@ MoneyFormatter getUiAmountAndNamePrinter() {
         return this.uiAmountAndNamePrinter;
     }
 
-    Currency getDefaultCurrency() {
-        if (campaign != null) {
-            HashMap<String, Currency> possibleCurrencies = new HashMap<>();
+    synchronized Currency getDefaultCurrency() {
```

Comment:
Nice idea saving the currency for the day!

Reply:
What happens if the user moves around between systems (e.g. using gm move?) We probably want to invalidate the cached value in that case as well.

Reply:
I'll invalidate the cache for that too, good catch.